experimental/gracefulblobmissing/patches.py

```diff
@@ -238,9 +260,37 @@ def patched_loadBlob_relstorage(self, cursor, oid, serial):
         lock.close()
 
 
-def create_empty_blob(filename):
-    mkdir("-p", '/'.join(filename.split('/')[:-1]))
-    fo = open(filename, 'w')
-    fo.write("File created by experimental.gracefulblobmissing.")
-    fo.close()
-    logger.info("Created missing blob-file %s" % filename)
+def guess_extension(blob=None):
+    ''' Try to guess the type of the blob out of thin air
+    '''
+    request = getRequest()
+    if not request:
+        return 'txt'
+    try:
+        published = repr(request.PUBLISHED.im_class)
+    except:
+        published = ''
+    if 'ImageScal' in published:
+        return 'png'
+    try:
+        last_parent = repr(request.other['PARENTS'][-1])
+    except:
+        last_parent = ''
+    if 'ImageScal' in last_parent:
+        return 'png'
+    return 'txt'
```

Comment:
The only two possible values that are returned here can be either "txt" or "png" - I wonder if the dummy blob.pdf file is really necessary for this PR.

Reply:
I agree. I would also skip the 'txt' version, and simply always use PNG.
In my case the detection fails, I get the txt version, and then I get errors because image scales cannot be generated.